src/panels/config/helpers/dialog-helper-detail.ts

```diff
@@ -0,0 +1,170 @@
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import "../../../components/ha-dialog";
+import { HomeAssistant } from "../../../types";
+import { dynamicElement } from "../../../common/dom/dynamic-element-directive";
+import { createInputBoolean } from "../../../data/input_boolean";
+import { createInputText } from "../../../data/input_text";
+import { createInputNumber } from "../../../data/input_number";
+import { createInputDateTime } from "../../../data/input_datetime";
+import { createInputSelect } from "../../../data/input_select";
+import { isComponentLoaded } from "../../../common/config/is_component_loaded";
+import { Helper } from "./ha-config-helpers";
+
+import "./forms/ha-input_boolean-form";
+
+const HELPERS = {
+  input_boolean: createInputBoolean,
+  input_text: createInputText,
+  input_number: createInputNumber,
+  input_datetime: createInputDateTime,
+  input_select: createInputSelect,
+};
+
+@customElement("dialog-helper-detail")
+export class DialogHelperDetail extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() private _item?: Helper;
+  @property() private _opened = false;
+  @property() private _platform?: string;
+  @property() private _error?: string;
+  @property() private _submitting = false;
+
+  public async showDialog(): Promise<void> {
+    this._platform = undefined;
+    this._item = undefined;
+    this._opened = true;
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
+    this._opened = false;
+    this._error = "";
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <ha-dialog
+        .open=${this._opened}
+        @closing=${this.closeDialog}
+        .heading=${this._platform
+          ? this.hass.localize(
+              "ui.panel.config.helpers.dialog.add_platform",
+              "platform",
+              this.hass.localize(
+                `ui.panel.config.helpers.types.${this._platform}`
+              ) || this._platform
+            )
+          : this.hass.localize("ui.panel.config.helpers.dialog.add_helper")}
+      >
+        ${this._platform
+          ? html`
+              <div class="form" @value-changed=${this._valueChanged}>
+                ${this._error
+                  ? html`
+                      <div class="error">${this._error}</div>
+                    `
+                  : ""}
+                ${dynamicElement(`ha-${this._platform}-form`, {
+                  hass: this.hass,
+                  item: this._item,
+                })}
+              </div>
+              <mwc-button
+                slot="primaryAction"
+                @click="${this._createItem}"
+                .disabled=${this._submitting}
+              >
+                ${this.hass!.localize("ui.panel.config.helpers.dialog.create")}
+              </mwc-button>
+            `
+          : html`
+              <div class="container">
+                ${Object.keys(HELPERS).map((platform: string) => {
```

Comment:
We can do better here. Should we have example widgets? Just empty ones not hooked up to anything.